### ISSUE TITLE
awsx:ecr:Image does not respect --target

### ISSUE THREAD

**Issue by sarink** [pulumi/pulumi#12194]
### What happened?

Running: `pulumi up -s my/stack -t **whatever**`

No matter what I place in `-t`, my `awsx.ecr.Image` is always built.

It doesn't actually attempt to _deploy_ the image, it just unnecessarily slows down the process in the worst way. Building docker images is usually one of the most time-consuming tasks, and it is precisely the thing I'm trying to avoid/the whole reason I'm using `--target` :-/ 

### Expected Behavior

I expect the image to not be built, unless it matches the `target` parameter

### Steps to reproduce

Actual example:

```
    const repo = new awsx.ecr.Repository('api-ecr-repo', { forceDelete: true });

    const apiImage = new awsx.ecr.Image('api-docker-image', {
      path: '../../',
      dockerfile: '../../api/Dockerfile',
      repositoryUrl: repo.url,
    });
```

Command: 
```
$ pulumi up -s org/prod -t **lambda**
```

Output:
```
     Type                               Name                                   Plan        Info
     pulumi:pulumi:Stack                prod                                   running
 +   ├─ aws:ec2:SecurityGroup           api-lambda-security-group              create      
     ├─ awsx:ecr:Image                  api-docker-image                                   Building image '../../'...
 +   ├─ aws:lambda:Function             api-lambda-function                    create
```

### Output of `pulumi about`

`v3.55.0`

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Frassle** [comment]
I think this is because `awsx.ecr.Image` is a component resource not a custom resource and those aren't effected by --target.

Being an MLC we probably could make --target effect them as well, but it would be inconsistent with non-mlc component resources. 

Alternatively it might be possible to turn Image into a Custom resource that only builds the image when the asset changes.

cc @danielrbradley for ideas on that last part.

**Comment by danielrbradley** [comment]
This might also be solved by the docker image work which will make the underlying docker image a custom resource rather than a component.

@Frassle will custom resources within MLC components be skipped with the existing `--target` behaviour?

**Comment by Frassle** [comment]
> will custom resources within MLC components be skipped with the existing --target behaviour?

Yup they should be skipped as long as --target doesn't include them (or their parent or dependencies)
